units/host_monitoring_idle_termination.go

```diff
@@ -212,26 +224,25 @@ func (j *idleHostJob) checkAndTerminateHost(ctx context.Context, h *host.Host) e
 			"idle":              idleTime.String(),
 			"last_communicated": communicationTime.String(),
 		})
-		return nil
+		return 0, 0, nil
```

Comment:
I see what you tried to do with this refactor, but the name is a bit misleading since you do more than get times (checking payment) and you return the wrong times depending on validation. 

Before if you hit this case, you exited checkAndTerminateHost. How you aren't returning an error to checkAndTerminateHost, you're just continuing through the function with numbers that aren't technically accurate and with no error, so you're still continuing on, not exiting early. 

Since idleTime and communicationTime are quick math calculations using things on the host, I argue it doesn't need to be passed out from the helper function (if you really want to compute it in only one place, maybe computing them outside the helper and passing them in would be better?). And then the function can be more of a validation function that returns bool, error or something (bool indicating if it can be terminated or something).